documentation/AuthServer.md

```diff
@@ -34,7 +35,9 @@ The document types used by the authorization server include:
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `oauth2ClientType` | String | The type of clients that use different OAuth2 grant type. GPII supports 4 type of clients: <br>`nativeApp`: native GPII apps that use OAuth2 resource owner password credentials grant.<br>`webApp`: web applications.<br>`thirdPartyApp`: third party applications that GPII users can define privacy settings for. <br>`clientCredentialsApp`: applications that use OAuth2 client credentials grant to add new preferences on behalf of users, such as First Discovery Tool.| Must be one of these values: "nativeApp", "webApp", "thirdPartyApp", "clientCredentialsApp".|

```

Comment:
@simonbates, all your review comments have been addressed except splitting "authDecision" documents into "authCodeAuthorization" and "onboardedSolutionAuthorization" documents. Running into road blocks at making this change:

1. For the "Onboarded solution authorizations" document structure:
- It seems reasonable to remove "redirectURL" field. What do you think?
- How do access tokens be assigned for this type of client?

2. How to handle the use case of "Mobile Accessibility":

As I understood, GPII users can define privacy policy for these 2 client types: "authCodeAuthorization" and "onboardedSolutionAuthorization". 

However, it's not clear how the use case of the android app "Mobile Accessibility" should be handled. As we discussed with @amb26 in the IRC channel at [here](https://botbot.me/freenode/fluid-work/2017-06-23/?msg=87676778&page=1), the most suitable client type for "Mobile Accessibility" seems like "oauth2ResourceOwner". Meanwhile, users are allowed to define privacy policy for it. This is different from the other "oauth2ResourceOwner" use case: local flow managers, with which users don't have controls of their privacy policy.

We need a clear picture of the authorization workflow for  "Mobile Accessibility" to determine if an extra client type is needed for it. Or, can we ditch the support of "Mobile Accessibility" if it was a temporary solution for a C4A review.